### ISSUE TITLE
New message about requiring API key breaks pull request and does not report message.

### ISSUE THREAD

**Issue by shauntruelove** [cmu-delphi/covidcast#636]
Your new message "This request exceeded the rate limit on anonymous requests, which will be enforced starting 2023-06-21. To be exempt from this limit, authenticate your requests with a free API key, now available at https://api.delphi.cmu.edu/epidata/admin/registration_form. A temporary public key `TEMP-API-KEY-EXPIRES-2023-06-28` is available for use between now and 2023-06-28 to give you time to register or adapt your requests without this message continuing to break your systems." is added as a row in the metadata, which breaks calls within you R package in the `covidcast_meta` function. Further, the reported error is really hard to track down, and does not report your message on needing an API key to the user.

**Comment by krivard** [comment]
Hi! We are in week 3 of a gradual rollout period to require API keys for certain kinds of requests. This change and the rollout schedule were announced on the API mailing list, which we recommend all users join so that they can receive notice of things like this ahead of time.

We know that not all users join, and not all users check the status message in the responses to their API requests, so the rollout period was intentionally designed to slowly become more disruptive over time to ensure people had time to adapt their pipelines before API keys became required for their use cases.

An obfuscated error condition was not intentional. Does this match the R error you observed?

```
Error in `dplyr::mutate()`:
ℹ In argument: `min_time = adjust_dates(.data$min_time, .data$time_type)`.
Caused by error in `match.arg()`:
! 'arg' should be one of “day”, “week”
```

If not, please provide the error you encountered.

This particular approach (slamming the error message into the first row of the returned data frame) will be active for another 8 days only. After that, requests that exceed the limits will result in HTTP 4xx errors, which are handled more easily by the covidcast R package.

[Consult this reference issue in delphi-epidata for more information and links to documentation for the fix](https://github.com/cmu-delphi/delphi-epidata/issues/1183).

In the issue title you mentioned breaking a pull request. Can you provide more details on that?

**Comment by shauntruelove** [comment]
Hi Katie,
Yes, this is the error I got. The problem is I never got any message about the API request status. All we got was the error. I had to dig through your code to finally find the error message inside the meta data frame. Maybe this is because of how we call your R package functions within our own functions or something. 

Sorry, misleading title. Not a pull request, and API call. Here is what it looked like for us. I have since registered for an API and do not have any issues now:

`>     us_hosp <- get_covidcast_hhs_hosp(geo_level = "state",
+                                       limit_date = Sys.Date())
Rows: 3202 Columns: 4                                                                                                                                                         
── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: ","
chr (3): abbreviation, location, location_name
dbl (1): population

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Joining, by = "state_fips"
Error in `dplyr::mutate()`:
! Problem while computing `min_time = adjust_dates(.data$min_time, .data$time_type)`.
Caused by error in `match.arg()`:
! 'arg' should be one of “day”, “week”
Run `rlang::last_error()` to see where the error occurred.`

**Comment by krivard** [comment]
Yep, I agree that's not a useful error message at all, and I'm sorry it took so much digging to figure out what was going on. The covidcast R client v0.5.0 was tested with steady-state error conditions, but not with the rollout ones; that was a lack of forethought on our part. Thankfully, the problem is temporary.

**Comment by krivard** [comment]
Rollout is complete; these errors now look like this in 0.5.0:

```
> x<-covidcast_meta()
Error in .request(list(source = "covidcast_meta", format = "csv")) : 
  Too Many Requests (RFC 6585) (HTTP 429). Failed to fetch data from API.
```

or like this in 0.5.1:

```
> x<-covidcast_meta()
Error in `covidcast_meta()`:
! Rate limit exceeded when fetching data from API anonymously. See the "API keys" section of the `covidcast_signal()` documentation for information on registering for an API key.
ℹ Message from server:
ℹ Rate limit exceeded for anonymous queries. To remove this limit, register a free API key at https://api.delphi.cmu.edu/epidata/admin/registration_form
Run `rlang::last_trace()` to see where the error occurred.
```
